scripts/hypervisor-facts.py

```diff
@@ -5,88 +5,94 @@
-  failed_names = {'interfaces': [], 'description': []}
-  default_hypervisor = facts['vm_facts_default_hypervisor_host']
-
-  # Create list if it doesn't already exist
-  if 'xen_vman_vms' not in facts:
+# Create list if it doesn't already exist
+if 'xen_vman_vms' not in facts:
     facts['xen_vman_vms'] = []
+if 'vm_facts_move_hypervisors' not in facts:
+    facts['vm_facts_move_hypervisors'] = []
 
-  # Traverse every host defined in hostvars
-  for host in original_facts.keys():
+# Traverse every host defined in hostvars
+for host in original_facts.keys():
     # Ignore physical hosts
     if 'vm' not in original_facts[host]:
-      continue
-    hypervisor_for_vm = original_facts[host]['vm']['hypervisor_host'] if 'hypervisor_host' in original_facts[host][
-      'vm'] else default_hypervisor
-    if hypervisor_host != hypervisor_for_vm:
-      continue
-    # config is the vm dict of a specific VM host
+        continue
     config = original_facts[host]['vm']
+    # Limit hosts to those defined in vm_facts_limit_hosts
+    if 'vm_facts_limit_hosts' in facts and host not in facts['vm_facts_limit_hosts']:
+        continue
+    hypervisor_for_vm = config['hypervisor_host'] if 'hypervisor_host' in config else default_hypervisor
+    if hypervisor_host != hypervisor_for_vm and (
+            'pull_hypervisor_from' not in config or config['pull_hypervisor_from'] != hypervisor_host):
+        continue
+    # Limit hosts to those defined in vm_facts_limit_hosts
+    if 'vm_facts_limit_hosts' in facts and host not in facts['vm_facts_limit_hosts']:
+        continue
+    # config is the vm dict of a specific VM host
+
     config['name'] = host
     # Ignore manually defined vms
     if any((d['name'] == host and 'org' in config and d['org'] == config['org']) for d in facts['xen_vman_vms']):
-      continue
+        continue
     if 'storage_type' in config:
-      storage_type = config['storage_type']
+        storage_type = config['storage_type']
     elif 'vm_facts_default_storage_type' in facts:
-      storage_type = facts['vm_facts_default_storage_type']
+        storage_type = facts['vm_facts_default_storage_type']
     else:
-      storage_type = 'filesystem'
+        storage_type = 'filesystem'
     # Set precise connection type for xen, depending on general storage_type
     if storage_type == 'blockdevice':
-      config['storage_type'] = 'iscsi'
+        config['storage_type'] = 'iscsi'
     else:
-      config['storage_type'] = 'nfs'
+        config['storage_type'] = 'nfs'
 
     # Copy description if necessary
     if 'description' not in config:
-      if 'description' in original_facts[host]:
-        config['description'] = original_facts[host]['description']
-      else:
-        failed_names['description'].append(host)
+        if 'description' in original_facts[host]:
+            config['description'] = original_facts[host]['description']
```

Comment:
Please use 2 spaces for indention as before

Reply:
But now it is PEP8 complient, everyone wanted this?

Reply:
This can be done with an extra PR. It's realy annoying to review this with all the blanks changed.